### ISSUE TITLE
Don't apply global templates to system indices in 7.x

### ISSUE THREAD

**Issue by gwbrown** [elastic/elasticsearch#64613]
System indices will mostly [be created automatically](https://github.com/elastic/elasticsearch/issues/616560) in 8.0, and will provide settings and mappings directly from the plugin managing the index, rather than using templates. However, it is not clear to what degree the code for doing so will be backportable to 7.x due to BWC concerns, so it may be useful to prevent some of the issues that exist when using index templates for system indices, as most system indices do at time of writing.

We already [prevent global templates from applying to hidden indices](https://github.com/elastic/elasticsearch/blob/79c267f4fc012eaf515c75c4fc1997db3c6d299b/server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java#L341-L343) in order to prevent cases where global index templates set up by users inadvertently impact e.g. Watcher or SLM history indices. It would be a very small code change to apply this to system indices as well, and simply ignore any global templates when system indices are created.

However, this would *technically* break backwards compatibility in a minor version: If a user is relying on a global template to apply e.g. allocation constraints to system indices, this would break that use case. However, the only indices impacted would be those owned by the Stack or other plugins, and is much more likely to prevent accidental issues than to cause substantial breakage.

/cc @jimczi

**Comment by rudolf** [comment]
:+1: From Kibana

It's not a common case, but we had a Kibana SDH where it seems like a global template changed the `dynamic_template` of the Kibana index which then caused Kibana to run into https://github.com/elastic/elasticsearch/issues/64585. Kibana >= 7.9.0 no longer has any `dynamic: true` fields so this is probably not going to happen again. But there is always the risk that a user adds settings to a global template that then breaks a system index which is not what the user intended.

**Comment by gwbrown** [comment]
Following an internal email discussion, we've decided not to move forward with this due to BWC concerns - it is not uncommon to apply allocation rules using global templates, and breaking that in a minor version could easily lead to problems with critical system indices being allocated to e.g. nodes with very slow storage.

While it's possible that we'll revisit this idea once we've solidified allocation management for system indices, there are still BWC concerns around using a different mechanism to manage this in a minor version. Therefore, I'm going to close this issue. If we believe there is a way to transition without breaking BWC, we can revisit this.
